bench/clblas_benchmark.cpp

```diff
@@ -0,0 +1,359 @@
+      MemBuffer<ScalarT> bufdst1(context, size);
+      MemBuffer<ScalarT> bufdst2(context, size);
+      MemBuffer<ScalarT> bufdst3(context, size);
+      cl_event events[3];
+      flops = benchmark<>::measure(no_reps, size * 3, [&]() {
+        clblasXaxpy<ScalarT>::func(size, alpha, bufsrc1.dev(), 0, 1,
+                                   bufdst1.dev(), 0, 1, 1, context._queue(), 0,
+                                   NULL, &events[0]);
+        clblasXaxpy<ScalarT>::func(size, alpha, bufsrc2.dev(), 0, 1,
+                                   bufdst2.dev(), 0, 1, 1, context._queue(), 0,
+                                   NULL, &events[1]);
+        clblasXaxpy<ScalarT>::func(size, alpha, bufsrc3.dev(), 0, 1,
+                                   bufdst3.dev(), 0, 1, 1, context._queue(), 0,
+                                   NULL, &events[2]);
+        clWaitForEvents(3, events);
+        clReleaseEvent(events[0]);
+        clReleaseEvent(events[1]);
+        clReleaseEvent(events[2]);
+      });
+    }
+    return flops;
+  }
+
+  BENCHMARK_FUNCTION(blas1_bench) {
+    UNPACK_PARAM;
+    double flops;
+    {
+      ScalarT alpha(3.135345123);
+      ScalarT vr[4];
+      unsigned vi;
+      MemBuffer<ScalarT> buf1(context, size);
+      MemBuffer<ScalarT> buf2(context, size);
+      MemBuffer<ScalarT, CL_MEM_READ_ONLY> bufr(context, vr, 4);
+      MemBuffer<unsigned, CL_MEM_READ_ONLY> buf_i(context, &vi, 1);
+
+      MemBuffer<ScalarT, CL_MEM_HOST_NO_ACCESS> scratch[4]{
+          MemBuffer<ScalarT, CL_MEM_HOST_NO_ACCESS>(context, size),
+          MemBuffer<ScalarT, CL_MEM_HOST_NO_ACCESS>(context, size),
+          MemBuffer<ScalarT, CL_MEM_HOST_NO_ACCESS>(context, 2 * size),
+          MemBuffer<ScalarT, CL_MEM_HOST_NO_ACCESS>(context, 2 * size),
+      };
+
+      cl_event events[5];
+      flops = benchmark<>::measure(no_reps, size * 12, [&]() {
+        clblasXaxpy<ScalarT>::func(size, alpha, buf1.dev(), 0, 1, buf2.dev(), 0,
+                                   1, 1, context._queue(), 0, NULL, &events[0]);
+        clblasXasum<ScalarT>::func(size, bufr.dev(), 0, buf2.dev(), 0, 1,
+                                   scratch[0].dev(), 1, context._queue(), 0,
+                                   NULL, &events[1]);
+        clblasXdot<ScalarT>::func(size, bufr.dev(), 1, buf1.dev(), 0, 1,
+                                  buf2.dev(), 0, 1, scratch[1].dev(), 1,
+                                  context._queue(), 0, NULL, &events[2]);
+        clblasXnrm2<ScalarT>::func(size, bufr.dev(), 2, buf1.dev(), 0, 1,
+                                   scratch[2].dev(), 1, context._queue(), 0,
+                                   NULL, &events[3]);
+        clblasiXamax<ScalarT>::func(size, buf_i.dev(), 0, buf1.dev(), 0, 1,
+                                    scratch[3].dev(), 1, context._queue(), 0,
+                                    NULL, &events[4]);
+        clWaitForEvents(5, events);
+        for (int i = 0; i < 5; ++i) clReleaseEvent(events[i]);
```

Comment:
This code will no longer be there, replaced with `Event::wait(std::vector<Event> &&events)`.